backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -154,6 +242,46 @@ public void handleProofEvent(PresentationExchangeRecord proof) {
                         }));
     }
 
+    // handles all proof request
+    public void handleProofRequestEvent(PresentationExchangeRecord proof) {
+        partnerRepo.findByConnectionId(proof.getConnectionId()).ifPresentOrElse(
+                p -> {
+                    pProofRepo.findByPresentationExchangeId(proof.getPresentationExchangeId())
+                            .ifPresentOrElse(pProof -> {
+                                if (PresentationExchangeState.PROPOSAL_SENT.equals(pProof.getState()) &&
+                                        PresentationExchangeState.REQUEST_RECEIVED.equals(proof.getState()) &&
+                                        PresentationExchangeInitiator.SELF.equals(proof.getInitiator())) {
+                                    log.info(
+                                            "Present_Proof: state=request_received on PresentationExchange where initator=self, responding immediatly");
+                                    presentProof(proof);
+                                }
+
+                            }, () -> {
+                                // new presentationExchangeID
```

Comment:
The proof state is already checked in the handler, so it will always have that state here

Reply:
That is true, but it becomes very hard to know what checks were made by the caller as this is a public function. And the two checks are in different classes. 

We should probably make an entire class to handle 'present_proof' events, then I'd feel comfortable running with those kinds of assumptions cause some methods will be private.